Xcode/pkg-support/resources/CMake/sdl2_net-config.cmake

```
@@ -0,0 +1,32 @@
+# SDL2_net CMake configuration file:
+# This file is meant to be placed in Resources/CMake of a SDL2_net framework
+
+# INTERFACE_LINK_OPTIONS needs CMake 3.12
+cmake_minimum_required(VERSION 3.12)
+
+include(FeatureSummary)
+set_package_properties(SDL2_net PROPERTIES
+    URL "https://www.libsdl.org/projects/SDL_net/"
+    DESCRIPTION "SDL_net is an example portable network library for use with SDL."
+)
+
+set(SDL2_net_FOUND TRUE)
+
+string(REGEX REPLACE "SDL2_net\\.framework.*" "SDL2_net.framework" _sdl2net_framework_path "${CMAKE_CURRENT_LIST_DIR}")
+string(REGEX REPLACE "SDL2_net\\.framework.*" "" _sdl2net_framework_parent_path "${CMAKE_CURRENT_LIST_DIR}")
+
+
+if(NOT TARGET SDL2_net::SDL2_net)
+    add_library(SDL2_net::SDL2_net INTERFACE IMPORTED)
+    set_target_properties(SDL2_net::SDL2_net
+        PROPERTIES
+            INTERFACE_COMPILE_OPTIONS "-F;${_sdl2net_framework_parent_path}"
+            INTERFACE_INCLUDE_DIRECTORIES "${_sdl2net_framework_path}/Headers"
+            INTERFACE_LINK_OPTIONS "-F;${_sdl2net_framework_parent_path};-framework;SDL2_net"
```

Have you tested the co-operation of an `SDL2.framework` including a cmake config file and a `SDL2_net.framework` using these cmake config files?

I'm asking because I don't own a Mac.
See https://github.com/libsdl-org/SDL_ttf/pull/213#issuecomment-1152837242 for the reason why I am asking.

---

I compiled a framework of `SDL` at `HEAD` and tried it along a freshly built SDL_net framework and I observed a few things:
- With both frameworks in the same folder (namely `/Library/Frameworks`), the `-F` option is only set once.
- Linking fails because the `-framework` flag is only set once (`-framework SDL2 SDL2_net` where `-framework SDL2 -framework SDL2_net` was expected)
- With one framework in in `/Library/Frameworks` and the other in `$HOME/Library/Frameworks`, the flags are only correctly set for the one linked first: `-F /Library/Frameworks -framework SDL2 $HOME/Library/Frameworks SDL2_net`

So it would appear that the [option de-duplication](https://cmake.org/cmake/help/latest/command/add_compile_options.html#option-de-duplication) you mentioned in the SDL_ttf PR is indeed breaking compilation.

```suggestion
            INTERFACE_COMPILE_OPTIONS "SHELL:-F ${_sdl2net_framework_parent_path}"
            INTERFACE_INCLUDE_DIRECTORIES "${_sdl2net_framework_path}/Headers"
            INTERFACE_LINK_OPTIONS "SHELL:-F ${_sdl2net_framework_parent_path} -framework SDL2_net"
```
This suggestion would fix the linking issues (and compilation warnings). With both frameworks in the same folder, it means `-F /Library/Frameworks` appears twice in each commands, but this seems practically harmless. 

What do you think we should do?

---

Thanks for testing!
Good to know option de-duplication is applied always, even when using frameworks.
I played locally with this (using dummy options), and I think we should do the following:
```suggestion
            INTERFACE_COMPILE_OPTIONS "-F ${_sdl2net_framework_parent_path}"
            INTERFACE_INCLUDE_DIRECTORIES "${_sdl2net_framework_path}/Headers"
            INTERFACE_LINK_OPTIONS "-F ${_sdl2net_framework_parent_path};-framework SDL2_net"
```
We can allow option de-duplication to happen. We should only be careful to let CMake drop the correct options.

---

Unfortunately, this suggestion breaks linking with the following error:
```
clang: error: unknown argument: '-framework SDL2_net'
``` 

With the suggested changes, the arguments are quoted in the resulting command:
```
c++ ... -F /Library/Frameworks/ -framework SDL2 "-F /Library/Frameworks" "-framework SDL2_net" ...
```

By applying the same change to SDL, the `-F` option is correctly deduped, but the same error for the `-framework` arises:
```
c++ ... "-F /Library/Frameworks/" "-framework SDL2" "-framework SDL2_net" ...
clang: error: unknown argument: '-framework SDL2'
clang: error: unknown argument: '-framework SDL2_net'
```

---

You're right! It really needs the `SHELL:` prefix.
```suggestion
            INTERFACE_COMPILE_OPTIONS "SHELL:-F ${_sdl2net_framework_parent_path}"
            INTERFACE_INCLUDE_DIRECTORIES "${_sdl2net_framework_path}/Headers"
            INTERFACE_LINK_OPTIONS "SHELL:-F ${_sdl2net_framework_parent_path};SHELL:-framework SDL2_net"
```

---

Can you also simulate a scenario where there is a space in the `_sdl2net_framework_parent_path` path variable?
We may need to escape the variable with double quotes.

---

Your assumption was correct and the framework parent path does need to be quoted to work with a space in the path. This is addressed in 03f5378.